jax/experimental/jax2tf/jax2tf.py

```diff
@@ -3051,7 +3051,7 @@ def _cond(
     index: TfVal, *operands: TfVal, branches: Sequence[core.ClosedJaxpr]
 ) -> Sequence[TfVal]:
   # tf.cond needs lambdas with no arguments.
-  branches_tf = [
+  branches_tf: list[Callable[[], Any]] = [
```

Comment:
An alternative fix here would be to use two different local variable names, but the type declaration also statically enforces the expected outcome of the list comprehension.